### DISCUSSION TITLE
How to implement a jump function similar to Xcode

### DISCUSSION THREAD

**Discussion by Mx-Iris** [krzyzanowskim/STTextView#52] [Questions]
I want to hold down the command key like in Xcode and then the cursor will change, while the corresponding token will have a special selection rect, and finally be able to respond to mouse clicks.

**Comment by Mx-Iris** [comment]
Is this possible in STTextView?

**Comment by krzyzanowskim** [comment]
@Mx-Iris tell us how did you solve the problem?

**Comment by Mx-Iris** [comment]
@krzyzanowskim To be honest, I haven't solved it yet. The methods I can think of are not elegant enough. Do you have any good ideas?

**Comment by krzyzanowskim** [comment]
There's no support for that right now. We look for something akin to `STTextViewDelegate.textView(_:clickedOnLink:at:)`, but it works for any location or maybe some custom string attribute. Combined with the `NSAttributedString.Key.cursor` text attribute, that should deliver what you think.
